### PR TITLE
docs: Create USERS.md

### PR DESCRIPTION
**Pull request by saradhis**
Move the list of official users to USERS.md as the README is getting longer.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlike to be merged.
* [ ] Optional. I've added My organization is added to the README.
* [ ] I've signed the CLA and required builds are green.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,66 +62,6 @@ Argo Workflows is an open source container-native workflow engine for orchestrat
 * DinD (docker-in-docker)
 * Script steps
 
-## Who uses Argo?
-As the Argo Community grows, we'd like to keep track of our users. Please send a PR with your organization name.
```
Comment (simster7) [inline README.md]: Instead of removing this whole section, why not show a couple "featured" users and link to `USERS.md` from here under "See all users?"

Comment (saradhis) [reply README.md]: I added a link to the USERS file. "Featured" would be a subjective list, so I would like to avoid that.

File: README.md [hunk 2]
```diff
@@ -35,6 +35,9 @@ Argo Workflows is an open source container-native workflow engine for orchestrat
 * Easily orchestrate highly parallel jobs on Kubernetes.
 * Argo Workflows puts a cloud-scale supercomputer at your fingertips!
 
+## Who uses Argo Workflows?
+[Offcial Argo Workflows user list](USERS.md)
```
Comment (simster7) [inline README.md]: Tiny typo:
```suggestion
[Official Argo Workflows user list](USERS.md)
```

File: README.md (hunk 1 above)
Comment (simster7) [reply README.md]: Fair point!

**Pull request merged**